exercises/retree/canonical-data.json

```
@@ -0,0 +1,57 @@
+{
+    "exercise": "retree",
+    "version": "1.0.0",
+    "cases": [
+      {
+        "description": "Empty tree",
+        "property": "treeFromTraversals",
+        "input": {
+          "preorder": [],
+          "inorder": []
+        },
+        "expected": {}
+      },
+      {
+        "description": "Tree with one item",
+        "property": "treeFromTraversals",
+        "input": {
+            "preorder": ["a"],
+            "inorder": ["a"]
+        },
+        "expected": {"v":"a", "l": {}, "r": {}}
+      },
+      {
+        "description": "Core functionality with realistic size traversal",
+        "property": "treeFromTraversals",
+        "input": {
+            "preorder": ["a", "i", "x", "f", "r"],
+            "inorder": ["i", "a", "f", "x", "r"]
+        },
+        "expected": {"v": "a",
+            "l": {"v": "i", "l": {}, "r": {}},
+            "r": {"v": "x",
+                "l": {"v": "f", "l": {}, "r": {}},
+                "r": {"v": "r", "l": {}, "r": {}}
+            }
+        }
+      },
+      {
+        "description": "Traversals must be fully consumed. No extra values.",
```

Maybe keep this consistent with the description of the next, invalid test case:

```suggestion
        "description": "Reject traversals of different length",
```